From 0cfeb67d5493b6975961d4dc21306ab1f1661783 Mon Sep 17 00:00:00 2001
From: Yu Jiahua <yujiahua1@huawei.com>
Date: Mon, 2 Aug 2021 05:09:52 -0800
Subject: [PATCH 3309/4585] [Huawei] sched: Introcude config option
 SCHED_OPTIMIZE_LOAD_TRACKING

Offering: HULK
hulk inclusion
category: feature
bugzilla: 175551

--------------------------------

Introduce kernel config SCHED_OPTIMIZE_LOAD_TRACKING to prepare
for kernel scheduler loadtracking optimize.

Signed-off-by: Yu Jiahua <yujiahua1@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Patchwork Links: http://patchwork.huawei.com/project/hulk5.10/list/?series=17550
Signed-off-by: Zheng Zucheng <zhengzucheng@huawei.com>
---
 init/Kconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 4eb9e68dc9f4..8112931ae91b 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -2134,6 +2134,14 @@ endmenu		# General setup
 
 source "arch/Kconfig"
 
+config SCHED_OPTIMIZE_LOAD_TRACKING
+	bool "Optimize scheduler load tracking"
+	default n
+	help
+	  Optimize scheduler load tracking, when load balance is not important
+	  in system, we close some load tracking in tick and enqueue or dequeue
+	  task, in this way, we can save some unnecessary cpu overhead.
+
 config RT_MUTEXES
 	bool
 
-- 
2.17.1

